kiva/agg/src/graphics_context.i

```diff
@@ -839,15 +840,26 @@ namespace kiva {
                 """
                 from PIL import Image
 
+                FmtsWithDpi = ('jpg', 'png', 'tiff', 'jpeg')
                 FmtsWithoutAlpha = ('jpg', 'bmp', 'eps', "jpeg")
                 size = (self.width(), self.height())
                 fmt = self.format()
 
+                if pil_options is None:
+                    pil_options = {}
+
+                file_ext = filename[-3:].lower() if isinstance(filename, str) else ''
```

Comment:
It seems we are following what was done for `FmtsWithoutAlpha` here, but we assume the extension is only 3 characters.  We could use `filename.rpartition(".")[-1].lower()` instead?  I guess we only need it for extensions in `FmtsWithDpi` / `FmtsWithoutAlpha` so not really needed...
Feel free to ignore

Reply:
Yeah, I copied the code as it was. That said, you version using `rpartition` should do just fine.